### PR TITLE
Fix build when the compiler is built with --enable-force-safe-string flag

### PR DESCRIPTION
**Pull request by hhugo**
Note that the first commit just removes trailing white spaces everywhere.

### CODE REVIEW

**Comment by hhugo** [comment]
cc @trefis

File: caml_z.c [hunk 1]
```diff
@@ -954,7 +958,7 @@ CAMLprim value ml_z_to_bits(value arg)
   Z_ARG(arg);
   r = caml_alloc_string(size_arg * sizeof(mp_limb_t));
   Z_REFRESH(arg);
-  p = (unsigned char*) String_val(r);
+  p = Bytes_val(r);
```
Comment (pascal-cuoq) [inline caml_z.c]: This change does not appear to follow the spirit of this comment:

https://github.com/ocaml/ocaml/issues/7594#issuecomment-473071206

Comment (hhugo) [reply caml_z.c]: reverted

**Comment by pascal-cuoq** [comment]
The PR looks quite simple and risk-free now.

**Pull request merged**